opds_import.py

```diff
@@ -268,55 +275,46 @@ def import_from_feed(self, feed, even_if_no_author=False,
                 # to this item.
                 message = StatusMessage(500, "Local exception during import:\n%s" % traceback.format_exc())
                 status_messages[key] = message
-                # go into circulation_obs, and remove any on the same identifier, 
-                # so we don't try to make pools and works from it
-                if key in circulation_objs:
-                    del circulation_objs[key]
                 # clean up any edition might have created
                 if key in imported_editions:
                     del imported_editions[key]
+                # Move on to the next item, don't create a work.
+                continue
 
 
-        for key, circulationdata in circulation_objs.iteritems():
             pool = None
             work = None
-            associated_metadata = None
-            if metadata_objs[key]:
-                associated_metadata = metadata_objs[key]
-            obj = DataSource.lookup(self._db, circulationdata._data_source)
-            # Note: is data_source is not lendable, don't make an error message, 
-            # not creating a pool is expected response.
-            if obj.offers_licenses:
-                try:
-                    pool = self.import_pool_from_circulationdata(
-                        circulationdata, associated_metadata, cutoff_date
-                    )
-
-                    work = self.import_work(
-                        pool, even_if_no_author, immediately_presentation_ready
-                    )
-                    # if error or cutoff making license pool, don't make work
-                    # if error or cutoff making work, do make license pool
-                    # always make editions
-                    if pool:
-                        imported_pools[key] = pool
-                        if work:
-                            imported_works[key] = work
-                        else:
-                            error_message = "No Work created for CirculationData %r, for an unknown reason." % circulationdata
-                            self.log.warn(error_message)
-                            raise DBImportException(error_message)
-                except Exception, e:
-                    message = StatusMessage(500, "Local exception during import:\n%s" % traceback.format_exc())
-                    status_messages[key] = message
-            else:
-                self.log.debug(
-                    "DataSource does not offer licenses.  No LicensePool created for CirculationData %r, not attempting to create Work.", 
-                    circulationdata
-                )
-
+            try:
```

Comment:
I moved these to helper methods, and wrote tests that override the helper methods to throw exceptions. I didn't write any additional tests for the helper methods themselves, are there any you think we need?